### PR TITLE
[SPARK-12436][SQL] Now InferSchema should return NullType and ArrayType(NullType)

### PR DESCRIPTION
**Pull request by naveenminchu**
a.) Currently I have fix for InferSchema to return NullType and ArrayType(NullType)
b.) @rxin  Can you confirm second task on [SPARK-12346] is talking about
prepareJobForWrite(job: Job) in "/spark-sql_2.10/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala"

### CODE REVIEW

**Comment by rxin** [comment]
cc @yhuai

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala [hunk 1]
```diff
@@ -169,7 +169,7 @@ private[json] object InferSchema {
         None
       }
 
-    case NullType => Some(StringType)
+    case NullType => Some(NullType)
```
Comment (yhuai) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala]: I think you can just delete this case.

**Comment by yhuai** [comment]
@naveenminchu I think the next step is to remove those null type fields before we write data to parquet/orc. @liancheng Parquet does not have a null type, right?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala (hunk 1 above)
Comment (naveenminchu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala]: @yhuai  Now removed NullType case as suggested.

**Comment by yhuai** [comment]
Can you add tests and also make changes to the write path (parquet/orc)? Otherwise, when you write a json dataset with null type to parquet, it will fail.

**Comment by nchammas** [comment]
Is this PR really meant to modify 1,320 files?

**Pull request closed without merging**